frontend/www/js/omegaup/components/UpdatedDatePicker.vue

```diff
@@ -0,0 +1,90 @@
+<template>
+  <b-input-group size="16">
+    <!-- Editable Text Input -->
+    <b-form-input
+      v-model="stringValue"
+      :name="name"
+      :disabled="!enabled"
+      placeholder="YYYY-MM-DD"
+      :class="{ 'is-invalid': isInvalid }"
+      required
+      @change="onInputChange"
+    />
+    
+    <!-- Calendar Input -->
+    <b-form-datepicker
+      v-model="stringValue"
+      :min="minDateStr"
+      :max="maxDateStr"
+      button-only
+      right
+      :disabled="!enabled"
+      @input="onDateSelected"
+    />
+  </b-input-group>
+</template>
+
+<script lang="ts">
+import { Vue, Component, Watch, Prop } from 'vue-property-decorator';
+import T from '../lang';
+import * as time from '../time';
+import { BFormDatepicker, BFormInput, BInputGroup } from 'bootstrap-vue';
+
+@Component({
+  components: { BFormDatepicker, BFormInput, BInputGroup },
+})
+export default class UpdatedDatePicker extends Vue {
+  T = T;
+
+  @Prop({ default: '' }) name!: string;
+  @Prop() value!: Date;
+  @Prop({ default: true }) enabled!: boolean;
+  @Prop({ default: T.datePickerFormat }) format!: string;
+  @Prop({ default: false }) isInvalid!: boolean;
+  @Prop({ default: null }) min!: Date | null;
+  @Prop({ default: null }) max!: Date | null;
+
+  private stringValue: string = '';
```

Comment:
To avoid using magic values, can it be declared like this?
```suggestion
  private stringValue: string | null = null;
```

Reply:
Updated to null